drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1637,7 +1649,8 @@ static void eth0_iface_init(struct net_if *iface)
 			     NET_LINK_ETHERNET);
 
 	/* Initialize GMAC queues */
-	for (i = 0; i < GMAC_QUEUE_NO; i++) {
+	for (i = CHIPID_CIDR_VERSION(CHIPID->CHIPID_CIDR) == 0 ?
```

Comment:
I doubt you can order specific revisions of that CPU and few people will want to inspect their board to know which software to flash.

Reply:
Ok

Reply:
I'd suggest to rewrite it the boring, easy to read way: a var, an if, an unencumbered for as it was before.

Reply:
> I doubt you can order specific revisions of that CPU and few people will want to inspect their board to know which software to flash.

Even the marking on the chip is not really clear. I have a B marked two lines below the reference.
 
> The datasheet calls the A revision the "legacy version" and the B revision the "current variant". Eventually all new boards will need the initialization for the extra queues.

I check a few resellers, it seems they mostly sell the B version (at least it's clear when ordering a chip), though it's still possible to find A versions.

Now about the GMAC case, adding three queue without activating them waste 480 bytes of RAM, and that will be reduced to 192 bytes when PR #11888 is merged. For a chip with 384kB of RAM, I think it's acceptable.

It might be an issue if one day we decide to actively use these queues by allowing users to map them to traffic classification queues. Right now the driver already support 2 additional queues, so that's probably acceptable.